libproc/lustre.c

```diff
@@ -173,11 +209,24 @@ _packed_lustre_version (pctx_t ctx)
 {
     int ret = -1;
     int major = 0, minor = 0, patch = 0, fix = 0;
+    pctx_t ctx_sys = NULL;
 
-    if ((ret = proc_fs_lustre_version (ctx, &major, &minor, &patch, &fix)) < 0)
-        goto done;
+    ret = proc_fs_lustre_version(ctx, &major, &minor, &patch, &fix);
+
+    if (ret < 0) {
+        ctx_sys = proc_create("/sys");
+        if (!ctx_sys) {
+            ret = -ENOMEM;
```

Comment:
The function comment says it returns -1 on error. Seems like these gotos should just be returns since there's no cleanup needed.